### DISCUSSION TITLE
How do you make a scrollable column that also expands to take up all as much available space as it needs to display its content?

### DISCUSSION THREAD

**Discussion by ItsCubeTime** [zauberzeug/nicegui#785] [Q&A] [answered]
### Question

Hi, Im looking to recreate below vanilla HTML document with nicegui, but I cant quite seem to get it working. I want to have a scrollable column/div  that tries to take up as much space as available until the point that it can display all its content - and if it cant fit everything, display a scrollbar (preferably using `display; overlay;` for the scrollbar).

The end goal is to be able to have multiple "views" displayed side by side, each view scrollable independently if all content in the view cant fit in its designated area (which could be limited to a % of screen width or height, a specifically set width/height or a % of space available in the container its inside)

```html
<style>
    html,
    body {
        margin: 0;
        height: 100%;
        overflow: hidden
    }
</style>
<div> <!--Random div just to demonstrate that nesting doesnt break the effect-->
    <div
        style="display: flex; flex-direction: column; overflow-y:overlay; overflow-x:clip;max-height: 100%; max-width: 200px;">
        <p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p><p>hi</p>
    </div>
</div>
```
![image](https://user-images.githubusercontent.com/20190653/232669591-37b7808b-954b-4036-ae96-bc8800a81af5.png)
*Works as intended*

### So I had a go at reconstruction the above code with nicegui
but I can only seem to get it to display the scrollbar if the max-height is set to a fixed value (like in `px`) rather than a `%` value (like in my below failed attempt at making the scrollbar appear also when using % max-height).

```py
from nicegui import ui

ui.html('''
<style>
    html, body, nicegui-layout, q-page-container, q-page, desktop, app { margin: 0; height: 100%; overflow: hidden; }
</style''')
with ui.column(): # Random excessive wrapping column - ideally this element wouldnt break the effect without the need of styling it.
    outerCol = ui.element('div')
    with outerCol:
        outerCol.style(add="display: flex; flex-direction: column; overflow-y:overlay; overflow-x:clip;max-height: 100%; max-width: 200px;")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
ui.run(port=1234)
```
![image](https://user-images.githubusercontent.com/20190653/232673122-407934dd-90fb-49d1-b469-5df1c9221525.png)
*Here texts elements just disappears out of view when they cant all fit on screen. It  kinda feel like the elements inside a nicegui app thinks the page has unlimited height even despite my attempted styling here?*

## Any help or advice would be appreciated 💝

**Comment by falkoschindler** [comment, marked as answer]
Hi @ItsCubeTime, I played around with your code examples for a while, but couldn't find a solution without styling the wrapping column. It looks like Quasar does something to the `q-page` and/or `q-page-container` so that it behaves differently than your vanilla HTML page.

To process I'd suggest the following steps:
- Try to create an outer container that fills the screen without growing with its content. If in doubt, use the developer tools to check the computed size.
- I like to use faint background colors to see what is happening.
- Once you have such container, you can try creating a nested container that fills the available vertical height and scrolls on overflow.
- Note: `nicegui-layout`, `q-page-container` and `q-page` are classes, thus should be addressed like `.nicegui-layout` etc. in CSS. And `app` is an ID, thus should be addressed like `#add`. But ideally you don't have to modify their styling.
- I'd suggest to use `ui.add_head_html` for the style tag. Otherwise it is placed as an element and creates a gap below. This is the large white space at the top of your page.

**Comment by rodja** [comment]
Maybe it is possible with https://quasar.dev/layout/page-scroller?

**Comment by ItsCubeTime** [comment]
Was able to get it working with falkoschindler's approach.
```py
from nicegui import ui
outerOuter = ui.element('div')
with outerOuter:
    outerOuter.style(replace='width: calc(100vw - 100px); height: calc(100vh - 100px); background-color: red;')
    
    outerCol = ui.element('div')
    with outerCol: # @note Change overflow-x: overlay; to overflow-x: clip; if you only want vertical scrolling.
        outerCol.style(add="padding: 30px; display: flex; flex-direction: column; overflow-y:overlay; overflow-x:overlay; max-height: 100%; max-width: 200px;")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("fooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
        ui.label("foo")
ui.run(port=1235)
```
